### DISCUSSION TITLE
Pandas version

### DISCUSSION THREAD

**Discussion by isonimiguelBB** [sassoftware/saspy#365] [Q&A] [answered]
Which Pandas versions can I use with each saspy release?
Not found it the docs.

**Comment by tomweber-sas** [comment, marked as answer]
Hey, I'm not aware of any pandas version constraints. saspy supports at least back to Python 3.4. And the versions of Pandas from then till current work. saspy only uses a few methods, for creating or reading data frames. So, nothing I'm using it for now is dependent upon anything in a newer release than what was out with python 3.4, or probably much earlier. 
Tom

BTW, you're the first to post in this new 'Discussions' section! I've been wondering if this would get used. So, if I had a prize for first poster, you'd have won it, but I don't, sorry :)  LOL

**Reply by isonimiguelBB** [reply]
Thank you! I will test with pandas 1.2.2 and let you know!

**Reply by tomweber-sas** [reply]
Cool. I just looked at what version I have on my main development box. It's python 3.5 and here's pandas:

tom64-5> pip show pandas
Name: pandas
Version: 0.24.2
Summary: Powerful data structures for data analysis, time series, and statistics
Home-page: http://pandas.pydata.org
Author: None
Author-email: None
License: BSD
Location: /usr/lib64/python3.5/site-packages
Requires: numpy, pytz, python-dateutil
Required-by: fastparquet, pandas-highcharts, pipefitter
tom64-5>

That sounds a lot older than 1.2.2. But, unless they've broken core functionality (which I wouldn't expect), I would expect you're version to work just fine.

Ha, on windows I have (older, but python there is 3.6, which is newer ???):
C:\Users\sastpw>pip show pandas
Name: pandas
Version: 0.19.2
Summary: Powerful data structures for data analysis, time series,and statistics
Home-page: http://pandas.pydata.org
Author: The PyData Development Team
Author-email: pydata@googlegroups.com
License: BSD
Location: c:\programdata\anaconda3\lib\site-packages
Requires: python-dateutil, pytz, numpy

**Reply by tomweber-sas** [reply]
Just FYI:
A quick grep through my code show only these pandas methods, in the various sd2fd/df2sd methods:
DataFrame.from_records()
read_csv()
to_datetime()
to_numeric()
isnull()

and this on in the transpose method
DataFrame.from_dict()

So, there ought to be nothing special with these that wouldn't work with whatever version of pandas, unless they've done something crazy to break backward compatibility with any of these methods.

Thanks,
Tom
